### ISSUE TITLE
Timed Out kicks

### ISSUE THREAD

**Issue by Beelzebu** [SpigotMC/BungeeCord#2271]
Hi, after updating the bungeecord to 1.12.2 (build 1273) some players get disconnected while changing server with the reason "Timed Out", after looking the closed issues I find that #2263 and #2267 are about the same, all my spigot servers are running 1.12.2 (I update the spigot jar every day, so always have the latest version) with ProtocolSupport, I tried removing ProtocolSupport thinking that this can be the problem but is the same, seems that happens to players with a ping around 200-250 or a big packet loss.

**Comment by Dimatert9** [comment]
I also have this problem. In latest bungeecord players kicks less. 600 players vs 3000 playes before update.
But now it kicks after ~15 sec after server switch. Before update it kick instantly when player change server.
Why just not remove kick code from spigot? Before 1.12.2, there was no code that kicked the players and everything was fine.

**Comment by Dimatert9** [comment]
https://www.spigotmc.org/threads/logging-in.276793/
https://www.spigotmc.org/threads/players-timing-out-spamming-chat.275386/
https://www.spigotmc.org/threads/viabackwards.170195/page-10#post-2689454
https://www.spigotmc.org/threads/viabackwards.170195/page-10#post-2683380

**Comment by md-5** [comment]
We can't just remove the timeout, it was added to prevent server crash exploits.
No one has been able to pin down any specific issue without using plugins, and I'm convinced the Bungee / Server code is otherwise solid.

What I think is happening (from mentions of loading screen and packet loss [don't care about viaversion etc]) is that your servers / players are just laggy and you're noticing it now a lot sooner because the timeout was effectively cut from 30 seconds to 15 seconds.

I've changed the timeout on the server to 30 seconds, so hopefully this will cut down on complaints.

**Comment by JensenYoung** [comment]
@md-5 how did you change the timeout time?
im running my own server and a lot of people are have the disconnected screen but console says Timed out

**Comment by md-5** [comment]
It was a spigot changw

**Comment by JensenYoung** [comment]
@md-5 do i remove spigot or downgrade?

**Comment by md-5** [comment]
Wat
There's no bug with the timeout

**Comment by Black-Hole** [comment]
@JensenYoung Please use the community support of the Spigot forums for help configuring your server and your plugins.

**Comment by Golicraft** [comment]
I have the exact same issues. I am not sure of to fix this but will continue to try.

**Comment by Luky584** [comment]
I have the same problem, no idea what's the cause. It happens even with no plugins.
Those packet-loss problems/network issue seems to be the reason. When I ran standalone server nothing like that happened. 
When I am running through the Bungee there seems to be a problem.

I have recently discovered, that 90% of the players, that are getting kicked are using Wifi. As I said before, it has not happened on regular standalone server before, why it happens on bungee then?
